#! /usr/bin/env python


# Initialize ros and import parameter generator
PACKAGE='test_polishing'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# Add parameters
#       Name     Type    Level    Description     	Default 	Min   	Max
gen.add("convergenceScale",double_t,0,"Convergence rate scale",1.0,0.0,5.0)
gen.add("velocityLimit",double_t,0,"Velocity limit",0.15,0,1)
gen.add("attractorOffsetX",double_t,0,"Attractor offset in x direction",0.0,-0.5,0.5)
gen.add("attractorOffsetY",double_t,0,"Attractor offset in Y direction",0.0,-0.5,0.5)
gen.add("attractorOffsetZ",double_t,0,"Attractor offset in Z direction",0.0,-0.5,0.5)
gen.add("kp",double_t,0,"kp",0.01,0,0.05)
gen.add("ki",double_t,0,"ki",0.05,0,0.1)
gen.add("usePid",bool_t,0,"Activate pid for force tracking",True)
gen.add("targetForce",double_t,0,"Target force to apply",10.0,0.0,20.0)

# Generate files
exit(gen.generate(PACKAGE, "test_polishing", "attractive_params"))